docs/codeql/codeql-language-guides/using-flow-labels-for-precise-data-flow-analysis.rst

```diff
@@ -139,29 +139,24 @@ is a barrier guard blocking flow through the use of ``data`` on the right-hand s
 At this point we know that ``data`` has evaluated to a truthy value, so it cannot be ``null``
 anymore.
 
-Implementing this additional condition is easy. We implement a subclass of ``DataFlow::BarrierGuardNode``:
+Implementing this additional condition is easy. We implement a predicate with the following signature:
```

Comment:
I have never used the `BarrierGuard` module, so could someone check that this new example makes sense?

Reply:
I think it needs to be changed. The implementation of `BarrierGuard::getABarrierNode` already takes into account the uses of the SSA variable for which the guarded expression/node is a use, so the user only needs to define what the guard is, which node it's warding, and in which branch it's doing it.

I think a better example would be a simplified version of the typical `startsWith` guard:

```ql
predicate startsWithGuard(Guard g, Expr e, boolean branch) {
  g.(MethodAccess).getMethod().hasName("startsWith") and
  e = g.(MethodAccess).getQualifier() and
  branch = true
}
```

Note that I wrote a Java version, since the signature `guardChecksSig` differs by language, e.g. in Java:

```ql
signature predicate guardChecksSig(Guard g, Expr e, boolean branch);
```

Reply:
I agree that it doesn't make sense to use SSA in a barrier guard (that's part of what the barrier guard module is supposed to help you with).

Reply:
I only now noticed that the documentation example is for JavaScript, which is the only language where `DataFlow::BarrierGuardNode` is still a thing 🤦 

I've reverted the documentation change.